refactor!: Rename EnvironmentDataLoader to EnvironmentInterface (#501)
thousandbrainsproject/tbp.monty: docs/how-to-use-monty/tutorials/running-your-first-experiment.md

**In this particular experiment, `n_train_epochs` was set to 1, and `max_train_steps` was set to 1. This means a single epoch was run, with one matching step per episode**. In the next section, we go up a level from the model step to understand episodes and epochs.

> ## `Simulation Environment`

Reviewer: **suggestion**: Environment Interface (no monospace font)

In the config for first_experiment, there is a comment that marks the start of data configuration. Now we turn our attention to everything below that line, as this is where episode specifics are defined.

The environment interface class is the way we interact with a simulation environment. The objects within an environment are assumed to be the same for both training and evaluation (for now), hence only one (class, args) pairing is needed. Note however that object orientations, as well as specific observations obtained from an object, will generally differ across training and evaluation. The environment interface is where you collect data from in order to train a model and do inference.

> The environment interface is basically the API between whatever Monty should learn about and the model. Its job is to send actions from Monty to the environment and return observations (+initialize and reset the environment). Note that the next observation is decided by the last action, and the actions are selected by a `motor_system`. This motor system is shared by reference with the model. By changing the actions, the **model** controls what it observes next, just as you would expect from a sensorimotor system.

Reviewer: ```suggestion
The environment interface is basically the API between the environment and the model. Its job is to sample from the environment and return observations to the model (+initialize and reset the environment). Note that the next observation is decided by the last action, and the actions are selected by a `motor_system`. This motor system is shared by reference with the model. By changing the actions, the **model** controls what it observes next, just as you would expect from a sensorimotor system.
```
